src/connectors/impls/otel/client.rs

```diff
@@ -147,17 +147,26 @@ impl Sink for OtelSink {
             for value in event.value_iter() {
                 let err = if self.config.metrics && value.contains_key("metrics") {
                     let request = ExportMetricsServiceRequest {
-                        resource_metrics: metrics::resource_metrics_to_pb(Some(value))?,
+                        resource_metrics: ctx.bail_err(
```

Comment:
would a heper function that includes `value` in the error be helpful here? Something like

```rust
fn try_convert<T>(value: Value, converter: Fn<(Value) ->T>, err: &str) -> Result<T> {...}
```

Reply:
I think we already do that, the logging framework lets people set levels based on module, so if we classify invalid content as warnings or even info they can just opt out for them. Also we could send them to the error port, together with the work that Rebeca is doing that could solve the whole problem once an for all :)